### ISSUE TITLE
feat: Add a callback option to `vim.lsp.buf.code_action()`

### ISSUE THREAD

**Issue by luckasRanarison** [neovim/neovim#25259]
### Problem

It's currently impossible to get information about the selected code action using `vim.lsp.buf.code_action`.

### Expected behavior

Why would that be useful? 
It would allow us to play around with code actions, for example: get the last code action context and apply the same code action to all diagnostics with the same context in the current workspace, or get the last code action edits and reuse it by implementing it as a dot repeat.

**Comment by justinmk** [comment]
As discussed in https://github.com/neovim/neovim/pull/25260, we want a generic solution for this, not special-purpose callbacks sprinkled in numerous places.

- https://github.com/neovim/neovim/pull/22598
- https://github.com/neovim/neovim/pull/22598#issuecomment-1463367606

**Comment by luckasRanarison** [comment]
I agree, having a global hook system would be better. Looking forward to it.
